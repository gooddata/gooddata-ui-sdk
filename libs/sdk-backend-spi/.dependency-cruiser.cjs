const depCruiser = require("../../common/config/dep-cruiser/default.config");

options = {
    forbidden: [...depCruiser.DefaultRules, ...depCruiser.DefaultSdkRules],
    options: depCruiser.DefaultOptions,
};

module.exports = options;
