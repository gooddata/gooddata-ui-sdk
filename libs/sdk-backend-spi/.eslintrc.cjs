// (C) 2020 GoodData Corporation
module.exports = {
    plugins: ["sonarjs", "eslint-plugin-tsdoc", "regexp"],
    extends: [
        "@gooddata",
        "plugin:import-esm/recommended",
        "plugin:sonarjs/recommended",
        "plugin:regexp/recommended",
        "../../.eslintrc.js",
    ],
    overrides: [
        {
            files: ["*.ts", "*.tsx"],
            parser: "@typescript-eslint/parser",
            parserOptions: { tsconfigRootDir: __dirname, project: "tsconfig.json" },
        },
    ],
};
