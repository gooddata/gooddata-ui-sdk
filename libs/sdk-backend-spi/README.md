# Analytical Backend Service Provider Interface (SPI)

## License

(C) 2020-2021 GoodData Corporation

This project is under MIT License. See [LICENSE](LICENSE).
