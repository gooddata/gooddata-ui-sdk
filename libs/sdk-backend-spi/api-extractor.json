/**
 * Config file for API Extractor.  For more info, please visit: https://api-extractor.com
 */
{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../common/config/api-extractor/api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/esm/index.d.ts",
    "messages": {
        "extractorMessageReporting": {
            // we do not want to enforce this rule for now in this project, re-enable when dashboards are not alpha
            "ae-incompatible-release-tags": {
                "logLevel": "none",
                "addToApiReportFile": false
            }
        }
    }
}
