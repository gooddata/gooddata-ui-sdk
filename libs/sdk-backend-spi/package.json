{
    "name": "@gooddata/sdk-backend-spi",
    "version": "8.0.0-alpha.101",
    "author": "GoodData",
    "description": "GoodData Backend SPI abstraction interfaces",
    "repository": "https://github.com/gooddata/gooddata-ui-sdk/tree/master/libs/sdk-backend-spi",
    "license": "BSD-3-Clause",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "src",
        "LICENSE"
    ],
    "config": {
        "tslint": "./{src,test}/**/*.ts"
    },
    "scripts": {
        "clean": "rm -rf ci dist coverage *.log && jest --clearCache",
        "dev": "tsc -p tsconfig.dev.json --watch",
        "build": "rm -rf dist && tsc -p tsconfig.build.json && npm run api-extractor",
        "api-extractor": "mkdir -p api && [ -z \"${CI}\" ] && (api-extractor run -l) || (api-extractor run)",
        "test": "jest --watch",
        "test-once": "jest",
        "test-ci": "JEST_JUNIT_OUTPUT=./ci/results/test-results.xml jest --ci --config jest.ci.js",
        "tslint": "tslint -p . -t verbose $npm_package_config_tslint",
        "tslint-ci": "mkdir -p ./ci/results && tslint -p . -t checkstyle -o ./ci/results/tslint-results.xml $npm_package_config_tslint",
        "prettier-check": "prettier --check '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "prettier-write": "prettier --write '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "dep-cruiser": "depcruise --validate .dependency-cruiser.js --output-type err-long src/",
        "dep-cruiser-ci": "depcruise --validate .dependency-cruiser.js --output-type err-long src/",
        "validate": "npm run dep-cruiser && npm run tslint && npm run prettier-check",
        "validate-ci": "npm run dep-cruiser-ci && npm run tslint-ci && npm run prettier-check"
    },
    "dependencies": {
        "@gooddata/sdk-model": "^8.0.0-alpha.101",
        "lodash": "^4.17.15",
        "spark-md5": "^3.0.0",
        "ts-invariant": "^0.4.4",
        "tslib": "^2.0.0"
    },
    "devDependencies": {
        "@gooddata/tslint-config": "1.0.0",
        "@microsoft/api-extractor": "^7.3.8",
        "@types/jest": "^25.1.2",
        "@types/lodash": "^4.14.123",
        "@types/spark-md5": "^3.0.1",
        "dependency-cruiser": "^8.0.1",
        "jest": "^25.1.0",
        "jest-junit": "^3.0.0",
        "prettier": "~2.0.5",
        "ts-jest": "^25.2.0",
        "tslint": "^5.14.0",
        "tslint-config-prettier": "^1.18.0",
        "typescript": "3.9.5"
    }
}
