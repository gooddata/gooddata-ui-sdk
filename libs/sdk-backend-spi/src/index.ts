// (C) 2019-2024 GoodData Corporation
/**
 * This package provides definitions of the Service Provider Interface (SPI) for the Analytical Backend.
 *
 * @remarks
 * The interface defines functionality to be implemented for a particular backend to be used in GoodData.UI.
 * The Analytical Backend SPI for GoodData Cloud and GoodData.CN (joint codename `tiger` in `@gooddata/sdk-backend-tiger`) is almost fully implemented.
 *
 * @packageDocumentation
 */
export {
    IAnalyticalBackend,
    IAnalyticalBackendConfig,
    AnalyticalBackendFactory,
    prepareExecution,
    IAuthenticationContext,
    IAuthenticatedPrincipal,
    IAuthenticationProvider,
    NotAuthenticatedHandler,
} from "./backend/index.js";

export { IBackendCapabilities } from "./backend/capabilities.js";

export { IUserSettings, IWorkspaceSettings, IUserWorkspaceSettings } from "./common/settings.js";

export { IUserService } from "./user/index.js";
export { IUserSettingsService } from "./user/settings/index.js";

export {
    IExecutionFactory,
    IPreparedExecution,
    IExecutionResult,
    IDataView,
    ExplainConfig,
    IExplainResult,
    IExplainProvider,
    ExplainType,
} from "./workspace/execution/index.js";

export { IWorkspaceSettingsService } from "./workspace/settings/index.js";

export { CancelableOptions, ICancelable } from "./cancelation/index.js";

export {
    IGetInsightOptions,
    IGetVisualizationClassesOptions,
    IWorkspaceInsightsService,
    InsightOrdering,
    IInsightsQueryOptions,
    IInsightsQuery,
    IInsightsQueryResult,
    IInsightReferences,
    IInsightReferencing,
    InsightReferenceTypes,
    SupportedInsightReferenceTypes,
} from "./workspace/insights/index.js";

export {
    ExportDefinitionOrdering,
    IExportDefinitionsQuery,
    IExportDefinitionsQueryOptions,
    IExportDefinitionsQueryResult,
    IGetExportDefinitionOptions,
    IWorkspaceExportDefinitionsService,
    ExportDefinitionQuerySortDirection,
    ExportDefinitionQuerySortProperty,
    ExportDefinitionQuerySort,
} from "./workspace/exportDefinitions/index.js";

export {
    IElementsQueryFactory,
    IElementsQueryResult,
    IElementsQuery,
    IElementsQueryOptions,
    IElementsQueryAttributeFilter,
    IElementsQueryOptionsElementsByUri,
    IElementsQueryOptionsElementsByValue,
    IElementsQueryOptionsElementsByPrimaryDisplayFormValue,
    ElementsQueryOptionsElementsSpecification,
    isElementsQueryOptionsElementsByValue,
    isElementsQueryOptionsElementsByPrimaryDisplayFormValue,
    isValueBasedElementsQueryOptionsElements,
    IFilterElementsQuery,
    FilterWithResolvableElements,
} from "./workspace/attributes/elements/index.js";

export { IExportConfig, IExportResult } from "./workspace/execution/export.js";

export { IWorkspaceStylingService } from "./workspace/styling/index.js";
export {
    AnalyticalBackendError,
    NoDataError,
    DataTooLargeError,
    TimeoutError,
    ProtectedDataError,
    UnexpectedResponseError,
    UnexpectedError,
    NotSupported,
    NotImplemented,
    NotAuthenticated,
    NotAuthenticatedReason,
    LimitReached,
    ContractExpired,
    ErrorConverter,
    isAnalyticalBackendError,
    isNoDataError,
    isDataTooLargeError,
    isProtectedDataError,
    isUnexpectedResponseError,
    isUnexpectedError,
    isNotSupported,
    isNotImplemented,
    isNotAuthenticated,
    isLimitReached,
    isContractExpired,
    AuthenticationFlow,
    AnalyticalBackendErrorTypes,
} from "./errors/index.js";

export { IPagedResource } from "./common/paging.js";

export {
    IAnalyticalWorkspace,
    IWorkspacesQuery,
    IWorkspacesQueryFactory,
    IWorkspacesQueryResult,
    IWorkspaceDescriptor,
} from "./workspace/index.js";

export { IAttributeHierarchiesService } from "./workspace/attributeHierarchies/index.js";

export { IWorkspacePermissionsService } from "./workspace/permissions/index.js";

export { IWorkspaceAttributesService } from "./workspace/attributes/index.js";

export { IWorkspaceMeasuresService, IMeasureReferencing } from "./workspace/measures/index.js";

export { IWorkspaceFactsService } from "./workspace/facts/index.js";

export {
    IWorkspaceDashboardsService,
    IGetDashboardOptions,
    IGetDashboardPluginOptions,
    IGetScheduledMailOptions,
    IWidgetAlertCount,
    SupportedDashboardReferenceTypes,
    IWidgetReferences,
    SupportedWidgetReferenceTypes,
    IDashboardReferences,
    IDashboardWithReferences,
    IDashboardsQuery,
    IDashboardsQueryResult,
} from "./workspace/dashboards/index.js";
export {
    isDashboardLayoutEmpty,
    IWidgetWithLayoutPath,
    LayoutPath,
    layoutWidgets,
    layoutWidgetsWithPaths,
    walkLayout,
} from "./workspace/dashboards/utils.js";
export {
    IWorkspaceUsersQuery,
    IWorkspaceUsersQueryOptions,
    IWorkspaceUsersQueryResult,
} from "./workspace/users/index.js";
export {
    IDateFilterConfigsQuery,
    IDateFilterConfigsQueryResult,
} from "./workspace/dateFilterConfigs/index.js";

export {
    IWorkspaceCatalogFactory,
    IWorkspaceCatalogAvailableItemsFactory,
    IWorkspaceCatalog,
    IWorkspaceCatalogFactoryOptions,
    IWorkspaceCatalogWithAvailableItems,
    IWorkspaceCatalogWithAvailableItemsFactoryOptions,
    IWorkspaceCatalogFactoryMethods,
    IWorkspaceCatalogMethods,
} from "./workspace/ldm/catalog.js";

export { IWorkspaceDatasetsService } from "./workspace/ldm/datasets.js";

export {
    IMeasureExpressionToken,
    IObjectExpressionToken,
    IAttributeElementExpressionToken,
    ITextExpressionToken,
    ICommentExpressionToken,
    IBracketExpressionToken,
} from "./workspace/measures/measure.js";

export { IOrganization, IOrganizations } from "./organization/index.js";
export { IEntitlements } from "./entitlements/index.js";
export { IDataSourcesService } from "./dataSources/index.js";
export { ISecuritySettingsService, ValidationContext } from "./organization/securitySettings/index.js";
export { IOrganizationStylingService } from "./organization/styling/index.js";
export { IOrganizationSettingsService } from "./organization/settings/index.js";

export {
    IWorkspaceUserGroupsQuery,
    IWorkspaceUserGroupsQueryOptions,
    IWorkspaceUserGroupsQueryResult,
} from "./workspace/userGroups/index.js";

export { IWorkspaceAccessControlService } from "./workspace/accessControl/index.js";

export {
    IOrganizationUserService,
    IOrganizationUsersQuery,
    IOrganizationUsersQueryResult,
    IOrganizationUserGroupsQuery,
    IOrganizationUserGroupsQueryResult,
} from "./organization/users/index.js";
export { IOrganizationPermissionService, IPermissionsAssignment } from "./organization/permissions/index.js";
