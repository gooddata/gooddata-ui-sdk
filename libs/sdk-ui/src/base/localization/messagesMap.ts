// (C) 2007-2024 GoodData Corporation
import { translationUtils } from "@gooddata/util";

import enUS from "./bundles/en-US.json";
import deDE from "./bundles/de-DE.json";
import esES from "./bundles/es-ES.json";
import frFR from "./bundles/fr-FR.json";
import jaJP from "./bundles/ja-JP.json";
import nlNL from "./bundles/nl-NL.json";
import ptBR from "./bundles/pt-BR.json";
import ptPT from "./bundles/pt-PT.json";
import zhHans from "./bundles/zh-Hans.json";
import ruRU from "./bundles/ru-RU.json";
import itIT from "./bundles/it-IT.json";

/**
 * @internal
 */
export interface ITranslations {
    [key: string]: string;
}

/**
 * @internal
 */
export const messagesMap: { [locale: string]: ITranslations } = {
    "en-US": translationUtils.removeMetadata(enUS),
    "de-DE": deDE,
    "es-ES": esES,
    "fr-FR": frFR,
    "ja-JP": jaJP,
    "nl-NL": nlNL,
    "pt-BR": ptBR,
    "pt-PT": ptPT,
    "zh-Hans": zhHans,
    "ru-RU": ruRU,
    "it-IT": itIT,
};
