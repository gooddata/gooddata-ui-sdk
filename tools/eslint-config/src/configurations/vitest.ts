// (C) 2025 GoodData Corporation

import type { IConfiguration } from "../types.js";

const configuration: IConfiguration<"@vitest"> = {
    packages: [
        {
            name: "@vitest/eslint-plugin",
            version: "1.3.5",
        },
    ],
    plugin: "@vitest",
    extends: ["plugin:@vitest/legacy-recommended"],
    rules: {
        "@vitest/expect-expect": "off",
        "@vitest/no-commented-out-tests": "warn",
        "@vitest/valid-title": "warn",
        "@vitest/no-disabled-tests": "warn",
        "@vitest/no-focused-tests": "warn",
        "@vitest/no-identical-title": "warn",
        "@vitest/valid-expect": "warn",
    },
};

export default configuration;
