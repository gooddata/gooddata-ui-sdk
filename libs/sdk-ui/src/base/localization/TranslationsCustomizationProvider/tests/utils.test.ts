// (C) 2021-2022 GoodData Corporation

import {
    pickCorrectInsightWording,
    removeAllInsightToReportTranslations,
    removeAllWordingTranslationsWithSpecialSuffix,
} from "../utils.js";
import { IWorkspaceSettings } from "@gooddata/sdk-backend-spi";
import { ITranslations } from "../../messagesMap.js";
import { describe, expect, it } from "vitest";

const mockTranslation: ITranslations = {
    "mock.translation|insight": "Insight",
    "mock.translation|report": "Report",
};

describe("pickCorrectInsightWording", () => {
    const settings: IWorkspaceSettings = {
        workspace: "workspace",
    };

    it.each([
        ["insight", false],
        ["report", true],
    ])(
        "should return translations with %s when enableInsightToReport is set to %s",
        (_value: string, enableInsightToReport: boolean) => {
            const result = pickCorrectInsightWording(mockTranslation, {
                ...settings,
                enableInsightToReport,
            });

            expect(result).toMatchSnapshot();
        },
    );
});

describe("removeAllInsightToReportTranslations", () => {
    const mockTranslationWithoutExtraWords: ITranslations = {
        "mock.translation": "Insight",
    };
    it("should remove all insight to report words", () => {
        const result = removeAllInsightToReportTranslations({
            ...mockTranslationWithoutExtraWords,
            ...mockTranslation,
        });
        expect(result).toMatchObject(mockTranslationWithoutExtraWords);
    });
});

describe("removeAllWordingTranslationsWithSpecialSuffix", () => {
    const mockTranslationWithoutExtraWords: ITranslations = {
        "mock.translation": "Insight",
        "mock.translation.metric": "Measure",
    };

    it("should remove all insight to report words", () => {
        const result = removeAllWordingTranslationsWithSpecialSuffix({
            ...mockTranslationWithoutExtraWords,
            ...mockTranslation,
            "mock.translation.metric._measure": "Measure",
            "mock.translation.metric._metric": "Metric",
        });
        expect(result).toMatchObject(mockTranslationWithoutExtraWords);
    });
});
