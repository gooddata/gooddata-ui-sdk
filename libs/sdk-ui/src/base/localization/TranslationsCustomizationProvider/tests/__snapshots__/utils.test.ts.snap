// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pickCorrectInsightWording > should return translations with insight when enableInsightToReport is set to false 1`] = `
{
  "mock.translation": "Insight",
  "mock.translation|insight": "Insight",
  "mock.translation|report": "Report",
}
`;

exports[`pickCorrectInsightWording > should return translations with report when enableInsightToReport is set to true 1`] = `
{
  "mock.translation": "Report",
  "mock.translation|insight": "Insight",
  "mock.translation|report": "Report",
}
`;
